### PR TITLE
Pmd7 upgrade 7.2.0 remoting set2

### PR DESCRIPTION
**Pull request by jborgers**

### CODE REVIEW

File: src/main/resources/category/java/remoting.xml [hunk 1]
```diff
@@ -408,535 +408,584 @@ return HttpClientBuilder.create()
...
-(ancestor::ConstructorDeclaration//FormalParameter|ancestor::ClassOrInterfaceBody//FieldDeclaration[@Static=true()])[pmd-java:typeIs('com.fasterxml.jackson.databind.ObjectMapper')]//VariableDeclaratorId/@Name)
-]
-]]>
+    (: exclude method annotated with PostConstruct :)
+    //ClassOrInterfaceBodyDeclaration[count(Annotation//Name[@Image='PostConstruct']) = 0]
+    (: exclude factory where config is in method returning 'this' and one method, typically build(), returning objectMapper :)
+    /MethodDeclaration[count(Block//ReturnStatement/Expression/PrimaryExpression/PrimaryPrefix[@ThisModifier=true()])=0
+    or ancestor::ClassOrInterfaceBody[count(.//MethodDeclaration/ResultType/Type[pmd-java:typeIs('com.fasterxml.jackson.databind.ObjectMapper')]) != 1]]
+    (: modify operations on objectMapper in a method where objectMapper is a formal parameter or field :)
+    //Statement//Name[(concat(substring-before(@Image, '.config'), substring-before(@Image, '.set'), substring-before(@Image, '.register')
```
Comment (stokpop) [inline src/main/resources/category/java/remoting.xml:558]: FYI: I created a compact way for similar construction: 
```
and matches(@MethodName,"(set|put|clear|add|remove|replace)")
```

File: src/main/resources/category/java/remoting.xml [hunk 2]
```diff
@@ -408,535 +408,584 @@ return HttpClientBuilder.create()
...
+            in all requests for that connection.&#13;
+            Solution: Create/build HttpClient with proper connection pooling and timeouts once, and then use it for
+            requests. (jpinpoint-rules)
+        </description>
+        <priority>2</priority>
+        <properties>
+            <property name="tags" value="jpinpoint-rule,performance" type="String" description="classification"/>
+            <property name="xpath">
+                <value><![CDATA[
+//ClassDeclaration[count(.//Annotation[pmd-java:typeIs('org.springframework.context.annotation.Configuration')]) = 0]
```
Comment (stokpop) [inline src/main/resources/category/java/remoting.xml:643]: can use `not(pmd-java:hasAnnotation('...')`?

File: src/main/resources/category/java/remoting.xml [hunk 3]
```diff
@@ -408,535 +408,584 @@ return HttpClientBuilder.create()
...
+            http://github.com/jborgers/http-client-monitor helps here.&#13;
+            Solution: Use the HttpHost constructor with 2 (including port) or preferably 3 arguments (including port and
+            protocol). (jpinpoint-rules)
+        </description>
+        <priority>2</priority>
+        <properties>
+            <property name="tags" value="jpinpoint-rule,confusing,suspicious" type="String" description="classification"/>
+            <property name="xpath">
+                <value><![CDATA[
+//ConstructorCall[pmd-java:typeIs("org.apache.http.HttpHost")]/ArgumentList[@Size=1]/..
```
Comment (stokpop) [inline src/main/resources/category/java/remoting.xml:815]: can also place ArgumentList check inside brackets `[...]` to avoid `/..` at end?

**Review by stokpop — APPROVED** [review]

File: src/main/resources/category/java/remoting.xml (hunk 1 above)
Comment (jborgers) [reply src/main/resources/category/java/remoting.xml:558]: ah, die is handig ja. denk ik niet vaak aan regex gebruiken. Deze regel is is nog todo, neem ik dan mee.

File: src/main/resources/category/java/remoting.xml (hunk 2 above)
Comment (jborgers) [reply src/main/resources/category/java/remoting.xml:643]: good point, updated

File: src/main/resources/category/java/remoting.xml (hunk 3 above)
Comment (jborgers) [reply src/main/resources/category/java/remoting.xml:815]: yes, shorter, adjusted

**Pull request merged**